internal/pkg/consensus/power_table_view.go

```diff
@@ -32,56 +27,66 @@ func NewPowerTableView(q ActorStateSnapshot) PowerTableView {
 
 // Total returns the total storage as a BytesAmount.
 func (v PowerTableView) Total(ctx context.Context) (*types.BytesAmount, error) {
-	rets, err := v.snapshot.Query(ctx, address.Undef, address.StoragePowerAddress, power.GetTotalPower)
-	if err != nil {
-		return nil, err
-	}
+	// Dragons: re-write without using query
```

Comment:
Why the blanket removal of snapshot.Query abstraction instead of fixing that abstraction to work with the new vm?